libraries/standard/http/src/http_client.c

```diff
@@ -65,15 +65,15 @@ static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
     HTTPStatus_t returnStatus = HTTP_SUCCESS;
     uint8_t * pBufferCur = pRequestHeaders->pBuffer + pRequestHeaders->headersLen;
     size_t toAddLen = 0;
-    uint8_t hasTrailingLine = 0;
+    uint8_t hasTrailingLine = 0u;
 
     /* Backtrack before trailing "\r\n" (HTTP header end) if it's already written.
      * Note that this method also writes trailing "\r\n" before returning. */
     if( strncmp( ( char * ) pBufferCur - ( 2 * HTTP_HEADER_LINE_SEPARATOR_LEN ),
                  "\r\n\r\n", 2 * HTTP_HEADER_LINE_SEPARATOR_LEN ) == 0 )
     {
         /* Set this flag to backtrack in case of HTTP_INSUFFICIENT_MEMORY. */
-        hasTrailingLine = 1;
+        hasTrailingLine = 1u;
         pBufferCur -= HTTP_HEADER_LINE_SEPARATOR_LEN;
         pRequestHeaders->headersLen -= HTTP_HEADER_LINE_SEPARATOR_LEN;
```

Comment:
It is risky to change the value of `pRequestHeaders->headersLen` without actually changing the contents of the buffer. I suggest that you account for this overwriting in the happy-path branch that writes to the buffer.